### ISSUE TITLE
Research Tree fails to load properly when clicked

### ISSUE THREAD

**Issue by AbsoluteXeroEX** [zimberzimber/Ztarbound#4]
I've been working on a mod that has ZB integrated (for testing); I've updated the ZB core to the current commit (1.04i unless there's one I'm missing), but when attempting to access the research tree, I keep getting this error:

**[Error] Exception while invoking lua function 'treeSelected'. (LuaException) Error code 2, [string "/zb/researchTree/researchTree.lua"]:837: attempt to index a nil value (field '?')
stack traceback:
	[C]: in metamethod '__index'
	[string "/zb/researchTree/researchTree.lua"]:837: in global 'buildStates'
	[string "/zb/researchTree/researchTree.lua"]:470: in function <[string "/zb/researchTree/researchTree.lua"]:463>**

I've added new 'cutsomGridTileImages' and 'version' entries, both with appropriate strings and without in the research tree's .config file, tried creating a completely new .config file from scratch, and have even gone the route of having ZB loaded as a separate mod (with any/everything that would conflict with it removed from my own mod), but no matter what I do, I keep getting this same error.

(I can provide additional information if needed!)

**Comment by AbsoluteXeroEX** [comment]
Update - July 11, 2019 1:10 PM

So I double-checked back in my research tree config file, and found out what was causing that issue; turns out I had accidentally left some named 'children' with older names or had double letters where there weren't supposed to be. That's taken care of, but now I'm running into another issue where the tree will load, but it's just a single red line with no icons or any other paths; here's what I pulled from the log:

**[Error] Exception while invoking lua function 'update'. (LuaException) Error code 2, [string "/zb/researchTree/researchTree.lua"]:537: attempt to index a nil value (field '?')
stack traceback:
	[C]: in metamethod '__index'
	[string "/zb/researchTree/researchTree.lua"]:537: in global 'draw'
	[string "/zb/researchTree/researchTree.lua"]:616: in global 'pan'
	[string "/zb/researchTree/researchTree.lua"]:135: in function <[string "/zb/researchTree/researchTree.lua"]:128>**

My Lua-to-JSON isn't where I'd like it to be just yet, but I'm now out of ideas on where exactly to look to try and resolve this one (I also apologize if you're getting flooded with notifications because of this)

**Comment by AbsoluteXeroEX** [comment]
Okay, so it turns out that I'm just an idiot and didn't scrub my config file as thoroughly as I thought I did, and I now feel all the more embarrassed for having opened this issue.

TL;DR = closing due to ID-10-T error

**Comment by zimberzimber** [comment]
No worries 👍
